client.go

```diff
@@ -103,53 +104,68 @@ func httpRespWithRetries(c context.Context, req *Request) (*http.Response, error
 	}
 	var httpResp *http.Response
 	var err error
-	for i := 1; i <= req.maxAttempts; i++ {
+	for i := 1; ; i++ {
 		req.debugf("request attempt #%d", i)
 		httpResp, err = req.client.client.Do(reqc)
-		if err != nil {
+		if err != nil && err != io.EOF {
 			req.errorf("http.Client.Do err: %s | req: %s", err.Error(), req.String())
 			return nil, err
 		}
 
+		switch {
+		// returned when there is an underlying bad connection, so we want to retry as if it's a 500+ StatusCode
+		case err == io.EOF:
+			if !req.retryOnEOFError {
```

Comment:
I think this should be moved up to the initial error check. It feels strange to redo the check here and handle it the same as an error.